communication/src/main/java/org/jboss/da/communication/aprox/impl/AproxConnectorImpl.java

```diff
@@ -207,6 +210,44 @@ public RepositoryManipulationStatus removeRepositoryFromGroup(Repository reposit
         }
     }
 
+    @Override
+    /**
+     * Implementation note: dcheung tried to initially use HttpURLConnection
+     * and send a 'HEAD' request to the resource. Even though that worked,
+     * for some reason this completely makes Arquillian fail to deploy the testsuite.
+     * For that reason, dcheung switched to using a simple URL object instead with the
+     * try-catch logic.
+     *
+     * No dcheung doesn't usually talks about himself in the third person..
+     */
+    public boolean doesGAVExistInPublicRepo(GAV gav) throws CommunicationException {
+        StringBuilder query = new StringBuilder();
+
+        try {
+            DAConfig config = this.config.getConfig();
+            query.append(config.getAproxServer());
+            query.append("/api/group/public/");
+            query.append(gav.getGroupId().replace(".", "/")).append("/");
+            query.append(gav.getArtifactId()).append('/');
+            query.append(gav.getVersion()).append('/');
+            query.append(gav.getArtifactId()).append("-").append(gav.getVersion()).append(".pom");
+
+            URLConnection connection = new URL(query.toString()).openConnection();
+            try {
+                connection.getInputStream();
```

Comment:
I think this should be in try-with-resources (or close the InputStream in finally)